samza-azure/src/main/java/org/apache/samza/system/eventhub/admin/EventHubSystemAdmin.java

```diff
@@ -65,19 +65,12 @@ public EventHubSystemAdmin(String systemName, EventHubConfig eventHubConfig,
     this.eventHubClientManagerFactory = eventHubClientManagerFactory;
   }
 
-  private String getNextOffset(String currentOffset) {
-    // EventHub will return the first message AFTER the offset
-    // that was specified in the fetch request.
-    // If no such offset exists Eventhub will return an error.
-    return String.valueOf(Long.parseLong(currentOffset) + 1);
-  }
-
   @Override
   public Map<SystemStreamPartition, String> getOffsetsAfter(Map<SystemStreamPartition, String> offsets) {
-    Map<SystemStreamPartition, String> results = new HashMap<>();
-
-    offsets.forEach((partition, offset) -> results.put(partition, getNextOffset(offset)));
-    return results;
+    // In EventHubSystemConsumer#initializeEventHubsManagers, we exclude the offset that we specify. i.e.
```

Comment:
We only exclude the offset specified when the offset is set as the END_OF_STREAM constant. https://github.com/apache/samza/blob/032a16079db8cbb1caf00117b985f7575d2c823a/samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java#L285

Reply:
Synced with @lhaiesp offline, since `getOffsetsAfter` is only used for getting start offsets on restart (`StorageManagerUtil`, `OffsetManager`), it will have the same behavior as `START_OF_STREAM` in all current uses `getOffsetAfter`.